### PR TITLE
Change call to Lucene VectorSimilarityFunction.convertToScore

### PR DESCRIPTION
**Pull request by martin-gaievski**
Signed-off-by: Martin Gaievski <gaievski@amazon.com>

### Description
Replace call to VectorSimilarityFunction.convertToScore by custom implementation. Method convertToScore has been removed in [Lucene 9.3](https://github.com/apache/lucene/blob/branch_9_3/lucene/core/src/java/org/apache/lucene/index/VectorSimilarityFunction.java), comparing to earlier implementation in [Lucene 9.2](https://github.com/apache/lucene/blob/branch_9_2/lucene/core/src/java/org/apache/lucene/index/VectorSimilarityFunction.java#L108). This causes build failures like [this one](https://github.com/opensearch-project/k-NN/runs/7656569723?check_suite_focus=true)
I've done it in [2.x branch](https://github.com/opensearch-project/k-NN/commit/aebff805b0d47c9297d59741422d82e7dc0c8087), no backport needed, at that time "main" branch was ok.
  
### Check List
- [X] All tests pass
- [X] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/test/java/org/opensearch/knn/index/LuceneEngineIT.java [hunk 1]
```diff
@@ -43,6 +46,15 @@ public class LuceneEngineIT extends KNNRestTestCase {
...
     private static final float[][] TEST_QUERY_VECTORS = { { 1.0f, 1.0f, 1.0f }, { 2.0f, 2.0f, 2.0f }, { 3.0f, 3.0f, 3.0f } };
 
+    private static final Map<VectorSimilarityFunction, Function<Float, Float>> VECTOR_SIMILARITY_TO_SCORE = ImmutableMap.of(
+        VectorSimilarityFunction.EUCLIDEAN,
+        (similarity) -> 1 / (1 + similarity),
+        VectorSimilarityFunction.DOT_PRODUCT,
+        (similarity) -> (1 + similarity) / 2,
+        VectorSimilarityFunction.COSINE,
+        (similarity) -> (1 + similarity) / 2
+    );
```
Comment (vamshin) [inline src/test/java/org/opensearch/knn/index/LuceneEngineIT.java:56]: Are these functions correct? For cosine, is it not ```1/(1 + similarity)```?

Comment (martin-gaievski) [reply src/test/java/org/opensearch/knn/index/LuceneEngineIT.java:56]: I took it from [Lucene implementation](https://github.com/apache/lucene/blob/branch_9_2/lucene/core/src/java/org/apache/lucene/index/VectorSimilarityFunction.java#L72-L74). Seems with current implementation IT tests are passing for cosine - https://github.com/opensearch-project/k-NN/blob/main/src/test/java/org/opensearch/knn/index/LuceneEngineIT.java#L56

Comment (martin-gaievski) [reply src/test/java/org/opensearch/knn/index/LuceneEngineIT.java:56]: seems all tests are green

**Review by junqiu-lei — APPROVED** [review]

**Review by vamshin — COMMENTED** [review]
LGTM! Thanks

**Review by vamshin — APPROVED** [review]

**Pull request merged**